engine/common/insight_log/resource_insight_logs.py

```diff
@@ -0,0 +1,132 @@
+import enum
+import json
+import logging
+import re
+from abc import ABC, abstractmethod
+
+from django.apps import apps
+
+insight_logger = logging.getLogger("insight_logger")
+logger = logging.getLogger(__name__)
+
+
+class EntityEvent(enum.Enum):
+    CREATED = "created"
+    UPDATED = "updated"
+    DELETED = "deleted"
+
+
+class InsightLoggable(ABC):
+    @property
+    @abstractmethod
+    def public_primary_key(self):
+        pass
+
+    @property
+    @abstractmethod
+    def insight_logs_verbal(self) -> str:
+        """
+        insight_logs_verbal returns resource name for insight_log
+        """
+        pass
+
+    @property
+    @abstractmethod
+    def insight_logs_type_verbal(self) -> str:
+        """
+        insight_logs_type_verbal resource type for insight_log
+        """
+        pass
+
+    @property
+    @abstractmethod
+    def insight_logs_serialized(self) -> dict:
+        """
+        insight_logs_serialized returns resource, serialized for insight_log
+        """
+        pass
+
+    @property
+    @abstractmethod
+    def insight_logs_metadata(self) -> dict:
+        """
+        insight_logs_metadata returns resource metadata which should be present in the insight_log line
```

Comment:
can you please clarify the difference between this and insight_logs_serialized

Reply:
I think I got it, it is not used in the state comparison

Reply:
It's needed to enrich log line with data which should be always presented there, not only if state is differ